cmd/pod-scaler/frontend/src/app/Histograms/Histograms.tsx

```diff
@@ -0,0 +1,123 @@
+import {Buffer} from 'buffer';
+import * as React from 'react';
+import {Alert, Flex, Spinner} from '@patternfly/react-core';
+import {DeserializeHistogram, Histogram} from "@app/CircLLHist/CircLLHist";
+import {LogarithmicComparativePlot} from "@app/CircLLHist/LogarithmicComparativePlot";
+
+export interface HistogramsProps {
+    /** URL to fetch raw data from */
+    dataUrl: string;
+    /** Query parameters for the fetch */
+    parameters: string;
+}
+
+export interface rawData {
+    cutoff: string;
+    lower_bound: string;
+    merged: string;
+    histograms: string[];
+}
+
+export interface Data {
+    cutoff: number;
+    lower_bound: number;
+    merged: Histogram;
+    histograms: Histogram[];
+}
+
+export type HistogramData = Record<string, Data>;
+
+const process = (raw: Record<string, rawData>): HistogramData => {
+    const data: HistogramData = {};
+    for (const resource in raw) {
+            const datum: Data = {
+                cutoff: parseFloat(raw[resource].cutoff),
+                lower_bound: parseFloat(raw[resource].lower_bound),
+                merged: DeserializeHistogram(Buffer.from(raw[resource].merged, 'base64')),
+                histograms: [],
+            };
+            for (const histogram of raw[resource].histograms) {
+                datum.histograms.push(DeserializeHistogram(Buffer.from(histogram, 'base64')))
+            }
+            data[resource] = datum;
+    }
+    return data;
+}
+
+export const Histograms: React.FunctionComponent<HistogramsProps> = (
+    {
+        dataUrl,
+        parameters,
+    }: HistogramsProps) => {
+    const [data, setData] = React.useState<HistogramData>({});
+    const [fetchError, setFetchError] = React.useState<string>("");
+
+    React.useEffect(() => {
+        let mounted = true;
+        fetch(dataUrl + "?" + new URLSearchParams(parameters), {headers: {"Accept": "application/json"}}).then(async (res) => {
+            if (!res.ok) {
+                const raw = await res.text();
+                throw new Error(res.status + ": " + raw);
+            }
+            const raw = await res.json();
+            if (mounted) {
+                const processed = process(raw);
+                setData(processed);
+            }
+        }).catch((error) => {
+            if (mounted) {
+                setFetchError(String(error));
+            }
+        })
+        return () => {
+            mounted = false
+        };
+    }, [dataUrl, parameters]);
```

Comment:
You should make sure the `parameters` prop is the exact same object each render to avoid running the request again and again.

Reply:
How?

Reply:
They are stored in my redux store - so they *should* be? I think? Maybe?